webapp/packages/plugin-authentication-administration/src/Administration/Users/Teams/GrantedUsers/useGrantedUsers.tsx

```diff
@@ -49,10 +50,17 @@ export function useGrantedUsers(team: TeamInfo, mode: TeamFormMode): Readonly<St
         this.state.editing = !this.state.editing;
       },
       revoke(subjectIds: string[]) {
-        this.state.grantedUsers = this.state.grantedUsers.filter(subject => !subjectIds.includes(subject));
+        this.state.grantedUsers = this.state.grantedUsers.filter(subject => !subjectIds.includes(subject.userId));
       },
       grant(subjectIds: string[]) {
-        this.state.grantedUsers.push(...subjectIds);
+        this.state.grantedUsers.push(...subjectIds.map(id => ({ userId: id, teamRole: null })));
+      },
+      assignTeamRole(subjectId: string, teamRole: string | null) {
+        const user = this.state.grantedUsers.find(user => user.userId === subjectId);
+
+        if (user) {
+          user.teamRole = teamRole;
+        }
       },
```

Comment:
seems like we can mutate object in some closer component that have access to it, right now we need to propagate a lot of logic through components